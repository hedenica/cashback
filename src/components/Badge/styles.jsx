import styled, { css } from 'styled-components';

export const Container = styled.div`
  position: relative;
  border-radius: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 5px;

  ${({ size }) =>
    size === 'small' &&
    css`
      width: 80px;
      font-size: 14px;
    `}

  ${({ size }) =>
    size === 'medium' &&
    css`
      width: 110px;
      font-size: 15px;
    `}

  ${({ size }) =>
    size === 'large' &&
    css`
      width: 150px;
      font-size: 18px;
    `}

    ${({ variation }) =>
    variation === 'success' &&
    css`
      background: #c5f0de;
      color: #12945d;
    `}

    ${({ variation }) =>
    variation === 'warning' &&
    css`
      background: #ffeaa0;
      color: #806b21;
    `}

    ${({ variation }) =>
    variation === 'danger' &&
    css`
      background: #ffd1d1;
      color: #802323;
    `}

    @media (max-width: 425px) {
    font-size: 14px;
    min-width: 100px;
  }
`;
