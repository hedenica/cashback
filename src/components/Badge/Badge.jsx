/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './styles';

const Badge = ({ children, size, variation }) => (
  <Container size={size} variation={variation}>
    {children}
  </Container>
);

export default Badge;

Badge.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.string,
  variation: PropTypes.string,
};

Badge.defaultProps = {
  size: 'medium',
  variation: 'success',
};
