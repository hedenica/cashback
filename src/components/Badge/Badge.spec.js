import React from 'react';
import { render } from '@testing-library/react'

import Badge from './Badge'

describe('<Badge />', () => {
  it('should render the badge component with the default variation and size', () => {
    const { container } = render(
    <Badge>
      Aprovado
    </Badge>)

    expect(container).toMatchSnapshot();
  });

  it('should render the badge component with success variation and default size', () => {
    const { container } = render(
    <Badge variation="success">
      Aprovado
    </Badge>)

    expect(container).toMatchSnapshot();
  });

  it('should render the badge component with warning variation and default size', () => {
    const { container } = render(
    <Badge variation="warning">
      Em revis√£o
    </Badge>)

    expect(container).toMatchSnapshot();
  });

  it('should render the badge component with danger variation and default size', () => {
    const { container } = render(
    <Badge variation="danger">
      Reprovado
    </Badge>)

    expect(container).toMatchSnapshot();
  });

  it('should render the badge component with default variation and small size', () => {
    const { container } = render(
    <Badge size="small">
      Aprovado
    </Badge>)

    expect(container).toMatchSnapshot();
  });

  it('should render the badge component with default variation and large size', () => {
    const { container } = render(
    <Badge size="large">
      Aprovado
    </Badge>)

    expect(container).toMatchSnapshot();
  });
})
