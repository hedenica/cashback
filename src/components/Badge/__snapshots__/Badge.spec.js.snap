// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render the badge component with danger variation and default size 1`] = `
<div>
  <div
    class="sc-bdfBwQ broitX"
  >
    Reprovado
  </div>
</div>
`;

exports[`<Badge /> should render the badge component with default variation and large size 1`] = `
<div>
  <div
    class="sc-bdfBwQ kSTJok"
  >
    Aprovado
  </div>
</div>
`;

exports[`<Badge /> should render the badge component with default variation and small size 1`] = `
<div>
  <div
    class="sc-bdfBwQ bWwXU"
  >
    Aprovado
  </div>
</div>
`;

exports[`<Badge /> should render the badge component with success variation and default size 1`] = `
<div>
  <div
    class="sc-bdfBwQ dPlSLx"
  >
    Aprovado
  </div>
</div>
`;

exports[`<Badge /> should render the badge component with the default variation and size 1`] = `
<div>
  <div
    class="sc-bdfBwQ dPlSLx"
  >
    Aprovado
  </div>
</div>
`;

exports[`<Badge /> should render the badge component with warning variation and default size 1`] = `
<div>
  <div
    class="sc-bdfBwQ hEtXwh"
  >
    Em revis√£o
  </div>
</div>
`;
